### ISSUE TITLE
Add <env> name when configure

### ISSUE THREAD

**Issue by styk-tv** [madcore-ai/cli#67]
default from master, parameter specifies ANY branch, that branch will be used for all the repos local and remote, so this has to carry into cloud formation (env parameter) and all subsequent clones.

**Comment by botzill** [comment]
OK, so basically if we set the `development` this mean that when we start a new cluster node then it needs to be loaded from `development` as well? Also, this mean that DSL job needs to load plugins from `development` as well ?

**Comment by styk-tv** [comment]
entire chain, yes

if the branch specified in configure <branch> does not exist then pull should default to default master. for the most part we will do "development" test as we simply want to isolate dev test build from production.

in CF you can use the environment parameter, i think that was the idea initially but currently it serves no apparent purpose.

**Comment by botzill** [comment]
OK, but I think that we should have then not input branch but the env like: `prod, dev`? And we acn map env to specific branch. Let say that I give as input a branch that exists only in plugins repo? What should be the behaviour?

**Comment by styk-tv** [comment]
Well, we do it properly and we do it once. We should show in log the following sequence on Core and Plugins repos.

```
git clone -b <branch> <repo>
pushd (~/.madcore/<repo> on CLI) and (/opt/<repo> on Core/Node)
    git fetch
    git --no-pager log -50 --pretty=oneline
    git reset --hard <commit>
    git --no-pager log -1
popd
```

This should be used:

1.  In Cli
2. In Core during install 
3. In Core during any job pulls
4. In Any Nodes

In order to accomplish git reset under specific branch CLI needs to be aware of the following. CLI should NOT ask for those parameters at any time. CLI config should have these parameters present in config file but not set at all. These are only used when set before configure is run.
```
[core]
branch = master
commit = 10bc09364ab1778cddc2e48dbdd2b2d08db2c300

[plugins]
branch = development
commit = 452228e262db270e81d32bc07857468b634e0628
```

**Comment by botzill** [comment]
@styk-tv I don't really understand how this commit id are set in config ? As I understand, when we run `configure` with specific branch, we get the latest commit and save in config. Later when we make any other activities, like installing a plugin and creating a cluster, we start it using the latest commit we saved previously. This way we will use only the repos version we configured with.

So, every time we run `configure` we actually get the latest version of that branch and update the config, right ?

Just want to confirm that this is what you also have in mind.

Thx.
